cirq/ion/convert_to_ion_gates.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from cirq import ops, protocols, optimizers, circuits
+from cirq.ion import MS, two_qubit_matrix_to_ion_operations
+
+
+class ConvertToIonGates:
+    """Attempts to convert strange gates into IonGates.
+    """
+
+    def __init__(self, ignore_failures=False) -> None:
+        """
+        Args:
+            ignore_failures: If set, gates that fail to convert are forwarded
+                unchanged. If not set, conversion failures raise a TypeError.
+        """
+        super().__init__()
+        self.ignore_failures = ignore_failures
+
+    def convert_one(self, op: ops.Operation):
+        """
+        Convert a single (one- or two-qubit) operation
+        into ion trap native gates
+        :param op: gate operation to be converted
+        :return: the desired operation implemented with ion trap gates
+        """
+
+        # Known gate name
+        if isinstance(op, ops.GateOperation):
```

Comment:
The method does nothing if it's not a gate operation, instead of raising an error or etc.